src/Codegen/Constraints/ObjectBuilder.php

```diff
@@ -132,6 +148,29 @@ protected function getCheckMethodCode(
       $include_error_handling = true;
     }
 
+    $max_properties = $this->typed_schema['maxProperties'] ?? null;
+    $min_properties = $this->typed_schema['minProperties'] ?? null;
```

Comment:
someone asked about whether we should validate if these integers are non-negative. i noticed we don't validate that for `StringMinLengthConstraint` or `ArrayMinItemsConstraint` - is there a reason/context for that? 

if we were to add that check for objects, i'm trying to figure out where the best place for it would be so that it'd error out for the developer trying to generate a schema with negative min/max numbers. doesn't seem like the `getCheckMethod` is the right place.

Reply:
It wasn't intentional that we left those off, just didn't get around to it. I'd add it anywhere in this codegen path, maybe above where you set the class properties?

Reply:
okay, sounds good - i added checks for those above in the `build` method.